chalk-parse/src/parser.lalrpop

```diff
@@ -165,7 +165,7 @@ AssocTyValue: AssocTyValue = {
 };
 
 pub Ty: Ty = {
-    "for" "<" <l:Comma<LifetimeId>> ">" <t:Ty> => Ty::ForAll {
+    "for" "<" <l:Comma<LifetimeId>> ">" "fn" "(" <t:Ty> ")" => Ty::ForAll {
```

Comment:
we should probably also permit the syntax ``"fn" "(" ... ")"` with an empty list of lifetime ids